Feature: Move celery.default_queue to operators.default_queue to allow re-use between executors

leonsmith opened the pull request:
The `default_queue` config option resides in the `celery` section.

We already have an operators config section which would be a better place to specify a default.
This also allows other executors to re-use the default queue functionality without having to set celery specific config.

closes: https://github.com/apache/airflow/issues/14696
<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of existing issue, reference it using one of the following:

closes: #ISSUE 
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

leonsmith committed:
Move celery.default_queue to operators.default_queue to allow re-use between executors

leonsmith commented on airflow/configuration.py:
```diff
@@ -159,6 +159,7 @@ class AirflowConfigParser(ConfigParser):  # pylint: disable=too-many-ancestors
         ('metrics', 'statsd_datadog_tags'): ('scheduler', 'statsd_datadog_tags', '2.0.0'),
         ('metrics', 'statsd_custom_client_path'): ('scheduler', 'statsd_custom_client_path', '2.0.0'),
         ('scheduler', 'parsing_processes'): ('scheduler', 'max_threads', '1.10.14'),
+        ('operators', 'default_queue'): ('celery', 'default_queue', '2.1.0'),
```
I'm not sure what version should be put here, guessing this would have to go as a new minor?

mik-laj replied:
I have a feeling that it doesn't matter much now, because the release manager decides whether this change will be included in the next release.

ashb replied:
Please put 2.1 though, yes. Saves us having to change it later.

turbaszek reviewed (approved):
Looks reasonable to me

ashb commented on airflow/config_templates/default_test.cfg:
```diff
@@ -64,6 +64,7 @@ auth_backend = airflow.api.auth.backend.default
 
 [operators]
 default_owner = airflow
+default_queue = default
```
```suggestion
```

We don't need to include defaults here if it's the same default from airflow.cfg

leonsmith replied:
Removed as suggested, this applied to the old config value under the celery section (it was set to the same value as airflow.cfg)

ashb replied:
Yeah, removing makes sense, just no need to add it to the new section

ashb reviewed (commented):
(non blocking change request)

leonsmith committed:
Remove `default_queue` from airflow/config_templates/default_test.cfg

Remove `default_queue` from airflow/config_templates/default_test.cfg as its set to the same value from airflow.cfg

Co-authored-by: Ash Berlin-Taylor <ash_github@firemirror.com>

leonsmith committed:
Add in changes to UPDATING.md to reflect change in default_queue config location from celery to operators
